nixos/modules/hardware/iosched.nix

```diff
@@ -6,24 +6,53 @@
 }:
 
 let
+  inherit (lib)
+    concatLines
+    concatStringsSep
+    mapAttrsToList
+    optional
+    optionals
+
+    mkIf
+    mkOption
+    types
+    ;
+
   cfg = config.hardware.block;
   escape = lib.strings.escape [ ''"'' ];
+
   udevValue = types.addCheck types.nonEmptyStr (x: builtins.match "[^\n\r]*" x != null) // {
     name = "udevValue";
     description = "udev rule value";
     descriptionClass = "noun";
   };
 
-  inherit (lib)
-    mkIf
-    mkOption
-    types
-
-    concatLines
-    concatStringsSep
-    mapAttrsToList
-    optional
-    ;
+  udevRule =
+    {
+      rotational ? null,
+      include ? null,
+      exclude ? null,
+      scheduler,
+    }:
+    concatStringsSep ", " (
+      [
+        ''SUBSYSTEM=="block"''
+        ''ACTION=="add|change"''
+        ''TEST=="queue/scheduler"''
+      ]
+      ++ optionals (rotational != null) [
+        ''ATTR{queue/rotational}=="${if rotational then "1" else "0"}"''
```

Comment:
FYI: `toString`ing a boolean does precisely this.

Reply:
`builtins.toString false` returns `""` (empty string), not `"0"`.